### PR TITLE
Migrate from NSPs to KNPs

### PR DESCRIPTION
**Pull request by WadeBarnes**
Deployed and working in the OCP4 `dev` environment.

### CODE REVIEW

File: openshift/templates/agent/agent-deploy.yaml [hunk 1]
```diff
@@ -4,54 +4,123 @@ apiVersion: v1
...
-        - - ext:name=${NAME}${SUFFIX}
-        - - role=${CONTROLLER_ROLE}
-          - app=${APP_NAME}${SUFFIX}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE}-${TAG_NAME}
-        - - role=${WALLET_ROLE}
-          - app=${APP_NAME}${SUFFIX}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE}-${TAG_NAME}
+        Allow the controller to access the agent admin interface.
```
Comment (esune) [inline openshift/templates/agent/agent-deploy.yaml:46]: Nit.: should be controller and msg queue

File: openshift/templates/api/api-deploy.yaml [hunk 2]
```diff
@@ -7,40 +7,83 @@ metadata:
...
-          - $namespace=${NAMESPACE}-${TAG_NAME}
-        - - role=${AGENT_ROLE}
-          - app=${APP_NAME}${SUFFIX}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE}-${TAG_NAME}
-        - - role=${MSG_QUEUE_ROLE}
-          - app=${APP_NAME}${SUFFIX}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE}-${TAG_NAME}
+        Allow the agent and event processor to access the controller.
```
Comment (esune) [inline openshift/templates/api/api-deploy.yaml:23]: THis seems to allow agent and frontend traffic, not the event processor: is the event processor sending anything back to the controller - and would therefore need a rule here?

File: openshift/templates/backup/backup-deploy.yaml [hunk 3]
```diff
@@ -163,11 +169,16 @@ parameters:
     description: The name assigned to all of the resources defined in this template.
     required: true
     value: backup
+  - name: ROLE
+    displayName: Role
+    description: Role
+    required: true
+    value: backup
```
Comment (esune) [inline openshift/templates/backup/backup-deploy.yaml:176]: This may be for another PR: just wondering if we want to sync this configuration with the backup-container one and provide the NetworkPolicy that allows it to access dbs in this namespace by default just by adding the `backup=true` label to the backup targets.

File: openshift/templates/db/db-deploy.yaml [hunk 4]
```diff
@@ -6,6 +6,65 @@ metadata:
...
+    apiVersion: networking.k8s.io/v1
+    metadata:
+      name: ${NAME}${SUFFIX}
+      labels:
+        name: ${NAME}${SUFFIX}
+        app: ${APP_NAME}${SUFFIX}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the agent(s) to access the wallet.
```
Comment (esune) [inline openshift/templates/db/db-deploy.yaml:20]: Nit.: this should read "allow api, msg queue, schema spy, backup" to access the db

File: openshift/templates/msg-queue/msg-queue-deploy.yaml [hunk 5]
```diff
@@ -5,6 +5,75 @@ metadata:
...
+      name: ${NAME}${SUFFIX}
+      labels:
+        name: ${NAME}${SUFFIX}
+        app: ${APP_NAME}${SUFFIX}
+        app-group: ${APP_GROUP}
+        role: ${ROLE}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the agent and event processor to access the controller.
```
Comment (esune) [inline openshift/templates/msg-queue/msg-queue-deploy.yaml:47]: Nit.: Allow controller and msg queue worker to access the msg queue

**Review by esune — CHANGES_REQUESTED** [review]
A couple of tweaks we may want/need, other than that looks good.

We can also decide to deal with the tweaks in another issue/PR.

File: openshift/templates/backup/backup-deploy.yaml (hunk 3 above)
Comment (WadeBarnes) [reply openshift/templates/backup/backup-deploy.yaml:176]: Can be dealt with in a separate PR.

**Review by esune — APPROVED** [review]
